Week 3 PR #2 based on review comments

rwackes opened the pull request:
In this PR, I validate client-side input and clarify code structure based on comments left by my hosts on my Week 3 PR.
- I added a required tag to the fields of my form that I need my user to fill in. If a user leaves a required field empty and attempts to submit the comment form, it is not accepted by my application and the user is given an alert to fill out the empty required field. Screenshot: https://screenshot.googleplex.com/vZHnq6QbsH6
- I specified the type of input I am expecting so that if a user does not fill in a field in the expected format, it is not accepted by my application and they are given an alert to fill out the field in the correct format.

rwackes committed:
Validate client-side input and clarify code structure based on Week 3 PR Review comments.

vkkpng-g commented on portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -14,15 +14,15 @@
 
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.FetchOptions;
```
Nit: alphabetize the import

rwackes replied:
Oops I accidentally left that. Fixed it by putting it below Entity import line

vkkpng-g reviewed (approved):
Thank you for taking the time to putting the changes in a new pr...it probably wasn't trivial to make this PR so I appreciate the PR!

Looks good to me :)

wdi-gg reviewed (commented):
Thanks for creating new PR!
Can you point me to the code that validates client-side input and explain which cases are valid / invalid for your website?

rwackes commented:
The code that validates client-side input is in the comments.html. On lines 27, 30, 33, 34, and 39, I added "required" and changed to type=email for email input that requires users write valid data in those input fields. So, until those inputs are filled in, they are highlighted in red (meaning they are required and have invalid input). Until all required input fields are filled in (contact number is not required and can be left blank), if a user attempts to click submit button, it will not post their comment and a HTML drop down tag will appear the first input field that is invalid (from top to bottom). The type=email tag requires a user write text in the format of johnsmith@emailservice.
